powder_reduction.py

```diff
@@ -23,111 +44,346 @@ def powder_reduction(sample='sample.nxs', calibration=None, lambda_binning=(0.7,
-        mask = sc.groupby(cal['mask'], group='spectrum').any('detector')
+        mask = sc.groupby(cal_sample['mask'], group='spectrum').any('detector')
 
         # Compute spectrum groups from detector groups
-        g = sc.groupby(cal['group'], group='spectrum')
+        g = sc.groupby(cal_sample['group'], group='spectrum')
 
         group = g.min('detector')
 
         assert group == g.max('detector'), \
             "Calibration table has mismatching group for detectors in same spectrum"
 
-        sample.coords['group'] = group.data
-        sample.masks['mask'] = mask.data
+        sample_data.coords['group'] = group.data
+        sample_data.masks['mask'] = mask.data
 
     # Correct 4th monitor spectrum
     # There are 5 monitors for WISH. Only one, the fourth one, is selected for
     # correction (like in the real WISH workflow).
 
-    # Select fourth monitor
-    mon4_selected = sample.attrs['monitor4'].values
+    # Select fourth monitor and convert from tof to wavelength
+    mon4_lambda = sc.neutron.convert(sample_data.attrs['monitor4'].values, 
+                                     'tof', 'wavelength')
 
+    # Spline background
+    mon4_spline_background = bspline_background(mon4_lambda,
+                                                sc.Dim('wavelength'),
+                                                smoothing_factor=70)
+    
     # Smooth monitor
-    mon4_smooth = smooth_data.smooth_data(mon4_selected,
-                                          dim='tof',
-                                          NPoints=40)
+    mon4_smooth = smooth_data(mon4_spline_background,
+                              dim='wavelength',
+                              NPoints=40)
     # Delete intermediate data
-    del mon4_selected
+    del mon4_lambda, mon4_spline_background
 
     # Correct data
     # 1. Normalize to monitor
-    # Convert to wavelength (counts and monitor)
-    sample_lambda = sc.neutron.convert(sample, 'tof', 'wavelength')
-
-    mon_conv = sc.neutron.convert(mon4_smooth, 'tof', 'wavelength')
+    # Convert to wavelength (counts)
+    sample_lambda = sc.neutron.convert(sample_data, 'tof', 'wavelength')
 
     # Rebin monitors' data
     lambda_min, lambda_max, number_bins = lambda_binning
-    mon_rebin = sc.rebin(mon_conv,
+    mon_rebin = sc.rebin(mon4_smooth,
                          'wavelength',
                          sc.Variable(['wavelength'],
                                      unit=sc.units.angstrom,
-                                     values=np.linspace(lambda_min, lambda_max, num=number_bins)))
+                                     values=np.linspace(lambda_min,
+                                                        lambda_max,
+                                                        num=number_bins)))
     sample_lambda /= mon_rebin
 
-    del mon_rebin, mon_conv, sample
+    del mon_rebin, mon4_smooth
+
+    # 2. absorption correction
+    # TODO: adapt for different number of inputs
```

Comment:
Why not just unpack the dict (dict keys have to match Mantid argument names) below:
```
bsorption_correction(sample, lambda_binning, **absorp)
```
Then there is no "TODO" and it should be flexible and keep working irrespective of future changes?

Reply:
Or maybe even better: Just pass the dict to `absorption_correction` and let that do the unpacking?

Reply:
I just submitted a new version of the file to deal with different values for the absorption correction related to Sample and Vanadium.